### PR TITLE
Fix pep8 & test unpack_pdu_objects compare dicts

### PR DESCRIPTION
**Pull request by areski**
This PR cleanup the codebase and start to make it pep8 compliant.
Pay attention to the second commit as it changes 2 test methods in order to compare pdu objects by dictionary comparison.

### CODE REVIEW

**Comment by areski** [comment]
I just added an extra commit to remove the user of wildcard in the tests.py

File: smpp/pdu_builder.py [hunk 1]
```diff
@@ -207,27 +140,19 @@ def add_message_payload(self, value):
...
         mandatory_parameters = self.obj['body']['mandatory_parameters']
         mandatory_parameters['number_of_dests'] = number_of_dests
         mandatory_parameters['dest_address'] = [] + dest_address
 
-    def addDestinationAddress(self,
-            destination_addr,
-            dest_addr_ton = 0,
-            dest_addr_npi = 0,
-            ):
+    def addDestinationAddress(self, destination_addr, dest_addr_ton=0, dest_addr_npi=0, ):
```
Comment (hodgestar) [inline smpp/pdu_builder.py]: Odd trailing comma here.

**Comment by areski** [comment]
@hodgestar thanks, I just removed the comma

File: tests.py [hunk 2]
```diff
@@ -108,29 +122,28 @@ def assertDictEquals(self, dictionary1, dictionary2, depth=[]):
...
         """
         Take a dictionary, pack and unpack it and dump it as JSON correctly
         """
         pdu_index = 0
-        for pdu in pdu_objects:
+        for pdu_object in pdu_objects:
             pdu_index += 1
             padded_index = '%010d' % pdu_index
             print '...', padded_index
+            str_eval = re.sub('null', 'None', eval('pdu_asserts.pdu_json_'+padded_index))
```
Comment (hodgestar) [inline tests.py]: Should this `+` not have spaces around it?

File: tests.py [hunk 3]
```diff
@@ -108,29 +122,28 @@ def assertDictEquals(self, dictionary1, dictionary2, depth=[]):
...
             self.assertEquals(
-                    re.sub('\n *','',
-                        prettydump(unpack_pdu(pack_pdu(pdu)))),
-                    re.sub('\n *','',
-                        eval('pdu_asserts.pdu_json_'+padded_index)))
-
+                pdu.unpack_pdu(pdu.pack_pdu(pdu_object)),
+                eval(str_eval)
```
Comment (hodgestar) [inline tests.py:145]: Do we need the second `eval` here?

File: tests.py [hunk 4]
```diff
@@ -142,12 +155,11 @@ def test_pack_unpack_pdu_hex_strings(self):
...
             print '...', padded_index
+            str_eval = re.sub('null', 'None', eval('pdu_hex_asserts.pdu_json_'+padded_index))
             self.assertEquals(
-                    re.sub('\n *','',
-                        prettydump(unpack_hex(pdu_hex))),
-                    re.sub('\n *','',
-                        eval('pdu_hex_asserts.pdu_json_'+padded_index)))
-
+                unpack_hex(pdu_hex),
+                eval(str_eval)
```
Comment (hodgestar) [inline tests.py:161]: Same comments about the spaces around `+` and second `eval` here.

**Comment by hodgestar** [comment]
Left a couple of comments but looks great otherwise. Thanks for doing this clean up! :D

File: tests.py (hunk 3 above)
Comment (areski) [inline tests.py:145]: Yes we do, the first eval is to import the object ie pdu_asserts.pdu_json_0000000001, the second is to transform the string to a dictionary.

**Comment by areski** [comment]
You are more than welcome!

File: tests.py (hunk 3 above)
Comment (hodgestar) [inline tests.py:145]: Perhaps we should replace the first `eval` with `getattr(pdu_asserts, 'pdu_json_' + padded_index)`? It's only one fewer eval, but does make what's happening a bit clearer.

Comment (areski) [inline tests.py:145]: I cannot agree more :) just pushed a correction for that

**Comment by hodgestar** [comment]
:+1:

**Comment by hodgestar** [comment]
Merged. Thank you!

**Comment by areski** [comment]
Thanks!

**Pull request merged**